starlette/responses.py

```diff
@@ -221,10 +217,14 @@ async def stream_response(self, send: Send) -> None:
         await send({"type": "http.response.body", "body": b"", "more_body": False})
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
-        await run_until_first_complete(
-            (self.stream_response, {"send": send}),
-            (self.listen_for_disconnect, {"receive": receive}),
-        )
+        async with anyio.create_task_group() as task_group:
+
+            async def wrap(coro: typing.Callable[..., typing.Awaitable]) -> None:
+                await coro()
+                task_group.cancel_scope.cancel()
+
+            task_group.start_soon(wrap, partial(self.stream_response, send))
+            task_group.start_soon(wrap, partial(self.listen_for_disconnect, receive))
```

Comment:
you can save a task here
```suggestion
            await wrap(partial(self.listen_for_disconnect, receive))
```